src/dlfcn.c

```diff
@@ -45,9 +45,6 @@
 #endif
 #endif
 
-#ifdef SHARED
-#define DLFCN_WIN32_EXPORTS
-#endif

```

Comment:
Now I'm looking at this change and basically you moved above `#define DLFCN_WIN32_EXPORTS` from one place (file `dlfcn.c`) to two places (files `CMakeLists.txt` and `Makefile`). So what is the benefit to have this macro defined on two places instead of one?

Reply:
I used the standard pattern for exporting symbols, that is: the build system specifies a macro on building the dll to say export the symbols. cmake for example has build support for that on creating shared libraries (https://cmake.org/cmake/help/v3.0/prop_tgt/DEFINE_SYMBOL.html). See the following compile line:
> cd /home/ralf/src/dlfcn-win32-cross-build/src && /usr/bin/i686-w64-mingw32-gcc -DDLFCN_WIN32_EXPORTS
> -DDLFCN_WIN32_SHARED **-Ddl_EXPORTS**  -pipe -Wall -fexceptions -fno-omit-frame-pointer 
> -fno-optimize-sibling-calls --param=ssp-buffer-size=4 -mms-bitfields -gstabs   -o CMakeFiles/dl.dir/dlfcn.c.obj 
>  -c /home/ralf/src/dlfcn-win32/src/dlfcn.c

Line 49 does the same, but limits this support to only one file. If the sources for the dl library were split into different source files for easier maintenance, such a macro definition is needed for each source file. As long as this is not done, my changes are not necessary. 

But if the export related changes are not applied, the following line needs to be added to the cmake build system `set_target_properties(dl PROPERTIES DEFINE_SYMBOL "")` to remove the obsolete -Ddl_EXPORTS macro.